js/customize-snapshots.js

```diff
@@ -53,6 +54,15 @@
 					snapshot.addPendingToStatusControl();
 					snapshot.addTitleControl( section );
 					snapshot.addInspectChangesetControl( section );
+
+					/**
+					 * This is not ideal but is just a workaround to fix https://core.trac.wordpress.org/ticket/42686 for 4.9.0.
+					 */
+					section.expanded.bind( function( expanded ) {
```

Comment:
Why only upon `expanded`?

Reply:
I wanted to change it as late as possible so that any other thing is not affected because of this change. Like the button text should not be publish
https://github.com/xwp/wordpress-develop/blob/master/src/wp-admin/js/customize-controls.js#L7944-L7948

Also upon `expanded`,  because the user will change the status only when he expands the publish section , and that's the time when this `api.settings.changeset.currentUserCanPublish` needs to be `true`